tests/unit/hook_packages/terraform/test_hooks.py

```diff
@@ -0,0 +1,628 @@
+"""Test Terraform Hooks"""
+from subprocess import CalledProcessError
+from unittest import TestCase
+from unittest.mock import Mock, call, patch
+from parameterized import parameterized
+
+from samcli.hook_packages.terraform.hooks import (
+    AWS_LAMBDA_FUNCTION_PROPERTY_BUILDER_MAPPING,
+    PROVIDER_NAME,
+    _get_s3_object_hash,
+    _build_cfn_logical_id,
+    _get_property_extractor,
+    _build_lambda_function_environment_property,
+    _build_lambda_function_code_property,
+    _translate_properties,
+    _translate_to_cfn,
+    _map_s3_sources_to_functions,
+    TerraformHooks,
+)
+from samcli.lib.hook.exceptions import PrepareHookException
+from samcli.lib.utils.resources import (
+    AWS_LAMBDA_FUNCTION as CFN_AWS_LAMBDA_FUNCTION,
+)
+
+
+class TestPrepareHook(TestCase):
+    def setUp(self) -> None:
```

Comment:
can you add some test cases for the TF applications with Lambda Function, and the aws_s3_object resources.

Reply:
I am asking for a test case to test `_translate_to_cfn` function with a tf plan JSON that contains one Lambda function resource that contains S3 bucket, and S3 Key that are used in another S3_Object resource. I believe `test_translate_to_cfn_with_mapping_s3_source_to_function` is testing tf plan json that contains two lambda resources, one with S3 attributes, and one with filename attribute

Reply:
you were right, that test was supposed to test translating with mapping sources but I missed adding the s3 objects in the test data. Fixed now 🙏